test/mini_sql/builder_tests.rb

```diff
@@ -194,4 +194,26 @@ def test_to_sql_with_params
 
     assert_equal(builder.to_sql(id: 10, is_sale: true), sql)
   end
+
+  def test_exception_when_section_not_defined
+    builder = @connection.build("SELECT * FROM products /*where2*/").where('id = ?', 10)
+
+    err = assert_raises(RuntimeError) { builder.to_sql }
+    assert_match 'Not found section /*where*/', err.message
+  end
+
+  def test_sql_literal
+    builder = @connection.build("SELECT * FROM products /*product_where*/")
+    builder.sql_literal(product_where: 'WHERE id = 10')
+
+    assert_equal(builder.to_sql, 'SELECT * FROM products WHERE id = 10')
+  end
+
+  def test_sql_literal_for_builder
+    user_builder = @connection.build("SELECT * FROM users /*where*/").where('id = ?', 10)
+    builder = @connection.build("SELECT * FROM (/*user_table*/) AS t")
+    builder.sql_literal(user_table: user_builder)
+
+    assert_equal(builder.to_sql, 'SELECT * FROM (SELECT * FROM users WHERE (id = 10)) AS t')
+  end
```

Comment:
I think we should test / define behavior of:

```
builder = conn.build("select 1 /*where*/")
builder.sql_literal(where: "where 1 = 1")
```

```
builder = conn.build("select 1 /*where*/")
builder.where("1 = 1")
builder.sql_literal(where: "1 = 1")
```

```
builder = conn.build("select 1 /*where*/")
builder.sql_literal(where: "2 = 2")
builder.sql_literal(where: "1 = 1")
```

specifically behavior around clashes and stacking should be defined in the test suite. I am fine to just raise errors.

Reply:
good point, I thought it would be better to prevent the use of predefined literals

```ruby
if PREDEFINED_SQL_LITERALS.include?(name)
  raise "/*#{name}*/ is predefined, use method `.#{name}` instead `sql_literal`"
end
```